### PR TITLE
Installer modifications for support matrix for k8s and OS

### PR DESCRIPTION
**Pull request by bhagyashree-sarawate**
Hi sneha,
Please review
Attaching logs for install and uninstall for k8s single master 1.15 version

[K8s_logs_single_master_1_15.txt](https://github.com/hpe-storage/python-hpedockerplugin/files/3755314/K8s_logs_single_master_1_15.txt)

Regards
Bhagyashree Sarawate

### CODE REVIEW

File: ansible_3par_docker_plugin/install_hpe_3par_volume_driver.yml [hunk 1]
```diff
@@ -1,4 +1,30 @@
...
+  become: root
+  vars:
+    multimaster: false
+    oc_result: {}
+
+  tasks:
+    - name: Set multimaster flag as true when we have more than one master in hosts file
+      set_fact:
+        multimaster: true
+      when: groups['masters'] | count | int  > 1
```
Comment (sneharai4) [inline ansible_3par_docker_plugin/install_hpe_3par_volume_driver.yml]: Remove extra white space before >

File: ansible_3par_docker_plugin/install_hpe_3par_volume_driver.yml [hunk 2]
```diff
@@ -1,4 +1,30 @@
 ---
+- name: Verify Openshift and Kubernetes supported versions
```
Comment (sneharai4) [inline ansible_3par_docker_plugin/install_hpe_3par_volume_driver.yml]: Rename Openshift to OpenShift

File: ansible_3par_docker_plugin/tasks/check_kubernetes_prerequisite.yml [hunk 3]
```diff
@@ -0,0 +1,29 @@
+- name: Initialize default parameters
```
Comment (sneharai4) [inline ansible_3par_docker_plugin/tasks/check_kubernetes_prerequisite.yml]: Change to "Initialize Kubernetes default parameters"

File: ansible_3par_docker_plugin/tasks/check_openshift_prerequisite.yml [hunk 4]
```diff
@@ -0,0 +1,29 @@
+- name: Initialize Openshift default parameters
```
Comment (sneharai4) [inline ansible_3par_docker_plugin/tasks/check_openshift_prerequisite.yml]: Rename Openshift to OpenShift

File: ansible_3par_docker_plugin/tasks/configure_doryd_service.yml [hunk 5]
```diff
@@ -1,81 +1,52 @@
-  - name: Initialize multimaster, k8s_1_13, os_3_11 flags as false by default
-    set_fact:
-      multimaster: false
-      k8s_1_13: false
-      os_3_11: false
-      ks8_result: {}
-      result: {}
-      doryd_exists: false
+- name: Initialize default parameters
```
Comment (sneharai4) [inline ansible_3par_docker_plugin/tasks/configure_doryd_service.yml]: Initialize doryd default parameters

**Comment by bhagyashree-sarawate** [comment]
> 
> 
> @bhagyashree-sarawate Few minor changes wrt formatting given. Please check.
> 
> Rest all looks good.
> 
> Also, we need to add the table as part of installer doc:
> **Cluster** 	**OS 3.9** 	**OS 3.10** 	**OS 3.11** 	**K8S 1.11** 	**K8S 1.12** 	**K8S 1.13** 	**K8S 1.14** 	**K8S 1.15**
> **Single master** 	System process 	System process 	Deployment 	System Process 	System Process 	Deployment 	Deployment 	Deployment
> **Multi master** 	_Fail installer_ 	_Fail installer_ 	Deployment 	_Fail installer_ 	_Fail installer_ 	Deployment 	Deployment 	Deployment

@Sneha , done with the minior review comments

File: ansible_3par_docker_plugin/tasks/configure_doryd_service.yml [hunk 6]
```diff
@@ -1,81 +1,76 @@
...
+  shell: kubectl get pods --namespace kube-system -o wide | grep doryd
+  register: result
+  failed_when: false
+  changed_when: false
+  when: hostvars.localhost.doryd_as_container == true
+ 
+- name: Set flag doryd_exists as true if already running
+  set_fact:
+    doryd_exists: true
+  when: result is defined and result.rc == 0
```
Comment (sneharai4) [inline ansible_3par_docker_plugin/tasks/configure_doryd_service.yml]: @bhagyashree-sarawate  Wont this give issue that result.rc not defined?

File: ansible_3par_docker_plugin/tasks/create_etcd_service.yml [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+- name: Create service file
+  template:
+    src: ../files/etcd.service
+    dest: /etc/systemd/system/etcd_hpe.service
+    owner: root
+    group: root
+    mode: 0755
+  when: stat_result.stat.exists == false
+
+- name: Check etcd_hpe service is running
```
Comment (sneharai4) [inline ansible_3par_docker_plugin/tasks/create_etcd_service.yml]: @bhagyashree-sarawate  Why do we need to check this?

**Comment by sneharai4** [comment]
@bhagyashree-sarawate Please resolve conflicts.

File: ansible_3par_docker_plugin/README.md [hunk 8]
```diff
@@ -197,11 +202,24 @@ Once complete you will be ready to start using the HPE 3PAR Docker Volume Plug-i
...
+     $ cd python-hpedockerplugin/ansible_3par_docker_plugin
+     $ ansible-playbook -i hosts_standalone_nodes uninstall/uninstall_standalone_hpe_3par_volume_driver.yml
+     ```
+     
+     * Uninstall plugin along with etcd standalone environment:
+     ```
+     $ cd ~
+     $ cd python-hpedockerplugin/ansible_3par_docker_plugin
+     $ ansible-playbook -i hosts_standalone_nodes uninstall/uninstall_standalone_hpe_3par_volume_driver_etcd.yml
+     ```
```
Comment (sneharai4) [inline ansible_3par_docker_plugin/README.md]: @bhagyashree-sarawate Do add a pointer for Install plugin on standalone environment

**Comment by sneharai4** [comment]
@bhagyashree-sarawate Please resolve conflicts

File: README.md [hunk 9]
```diff
@@ -74,6 +74,8 @@ Troubleshooting issues with the plugin can be performed using these [tips](/docs
 - For upgrading the plugin from older version to the current released version, user needs to unmount all the volumes and follow the standard
  upgrade procedure described in docker guide. 
  
+- Encryption only supports in plugin version 3.1.1 and 3.3.1.
```
Comment (sneharai4) [inline README.md]: @bhagyashree-sarawate Update this to:
Encryption not working with plugin version 3.2 and 3.3.

File: docs/advanced/etcd_cluster_setup.md [hunk 10]
```diff
@@ -1,85 +1,143 @@
...
-
-
-[root@cssos196133 ~]# curl http://15.212.196.133:23790/v2/members
-{"members":[{"id":"9b0bff3d96b77fc3","name":"infra2","peerURLs":["http://15.212.196.135:23800"],"clientURLs":["http://15.212.196.135:23790","http://15.212.196.135:40010"]},{"id":"a3f74f32f40cc714","name":"infra0","peerURLs":["http://15.212.196.133:23800"],"clientURLs":["http://15.212.196.133:23790"]},{"id":"c5c61529df279a51","name":"infra1","peerURLs":["http://15.212.196.134:23800"],"clientURLs":["http://15.212.196.134:23790","http://15.212.196.134:40010"]}]}
-
-[root@cssos196133 ~]# curl http://15.212.196.133:23790/version
-{"etcdserver":"3.3.11","etcdcluster":"3.3.0"}[root@cssos196133 ~]#
+## List members of etcd cluster
+
+etcdctl --endpoint http://10.50.9.23:23790 member list
```
Comment (sneharai4) [inline docs/advanced/etcd_cluster_setup.md]: @bhagyashree-sarawate Please change the IPs to private IP or dummy IPs

**Pull request merged**